src/plugins/data_source_management/public/components/data_source_multi_selectable/data_source_multi_selectable.tsx

```diff
@@ -43,46 +47,49 @@ export class DataSourceMultiSelectable extends React.Component<
 
   async componentDidMount() {
     this._isMounted = true;
-    getDataSourcesWithFields(this.props.savedObjectsClient, ['id', 'title', 'auth.type'])
-      .then((fetchedDataSources) => {
-        if (fetchedDataSources?.length) {
-          // all data sources are selected by default on initial page load
-          const selectedOptions: SelectedDataSourceOption[] = fetchedDataSources.map(
-            (dataSource) => ({
-              id: dataSource.id,
-              label: dataSource.attributes?.title || '',
-              checked: 'on',
-              visible: true,
-            })
-          );
+    try {
```

Comment:
why change from `.catch()` to `try {} catch(){}`, I didn't see the code logic differer from before

Reply:
Hi Zhongnan this is the original code(https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source_management/public/components/data_source_multi_selectable/data_source_multi_selectable.tsx#L46). It puts all the logic under `if (fetchedDataSources?.length) {` I want to refactor the code, so i use the try{} catch {}
Now it is
```
try {
    const fetchedDataSources = await getDataSourcesWithFields....
   if (fetchedDataSources?.length) {
     ...
  }
   if (!this.props.hideLocalCluster) {
   ....
  }

}.catch() {
}
```

Reply:
@zhyuanqi got you. this is fixing the bug that local host not able to render due to datasource.length = 0. Thanks!